business/openshift_oauth.go

```diff
@@ -158,10 +157,15 @@ func getKialiNamespace() (string, error) {
 	return kialiNamespace, nil
 }
 
-func getKialiRoutePath(routeName string) (*string, error) {
+func getKialiRoutePath(config config.OpenShiftConfig) (*string, error) {
 	var route *OAuthRoute
 	var protocol string
 
+	if config.RedirectURL != "" {
```

Comment:
I never examined in detail this `getKialiRoutePath` func until now.

I wonder if this func may work here: https://github.com/kiali/kiali/blob/918476b417c67baadd3a017cfa5c0dc90bf4beaf/util/httputil/http_util.go#L181.

It would be to prevent adding a config option for the redirect URL. If it works, I may think this `getKialiRoutePath` may be completely replaced by that `GuessKialiURL` func.

Reply:
It seems to work that way as well, thanks!